app/javascript/mastodon/rtl.js

```diff
@@ -23,5 +23,14 @@ export function isRtl(text) {
     return false;
   }
 
+  // Remove mentions before counting characters to decide RTL ratio
+  text = text.replace(/@[0-9A-Za-z_@]+/g, '');
```

Comment:
Let me correct myself: When displaying toots from the server, rely on language.

When composing a new toot, we still need char-based RTL detection. Stripping out mentions and URLs is good, e-mails not worth it (as I said above), HTML tags unnecessary because they will not be in the compose form.